Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -800,24 +802,36 @@ public void Dispose()
         /// <summary>
         /// Generates the wallet file.
         /// </summary>
-        /// <param name="password">The password used to encrypt sensitive info.</param>
         /// <param name="name">The name of the wallet.</param>
-        /// <param name="extendedKey">The root key used to generate keys.</param>
+        /// <param name="encryptedSeed">The seed for this wallet, password encrypted.</param>
+        /// <param name="chainCode">The chain code.</param>
         /// <param name="creationTime">The time this wallet was created.</param>
         /// <returns></returns>
         /// <exception cref="System.NotSupportedException"></exception>
-        private Wallet GenerateWalletFile(string password, string name, ExtKey extendedKey, DateTimeOffset? creationTime = null)
+        private Wallet GenerateWalletFile(string name, string encryptedSeed, byte[] chainCode, DateTimeOffset? creationTime = null)
         {
+            Guard.NotEmpty(name, nameof(name));
+            Guard.NotEmpty(encryptedSeed, nameof(encryptedSeed));
+            Guard.NotNull(chainCode, nameof(chainCode));
+
             if (this.fileStorage.Exists($"{name}.{WalletFileExtension}"))
             {
-                throw new InvalidOperationException($"Wallet with name '{name}.{WalletFileExtension}' already exists.");
+                throw new InvalidOperationException($"Wallet with name '{name}' already exists.");
```

Comment:
This should probably be a WalletException no?

Reply:
we can make it that way yes :-)